tests/bluetooth/hci/testcase.yaml

```diff
@@ -0,0 +1,6 @@
+tests:
+  bluetooth.hci:
+    integration_platforms:
+      - native_sim
+    tags:
+      - bluetooth
```

Comment:
Should this also have `type: unit`?

Reply:
The test as it is now includes the entire zephyr stack. Now that it has moved to use only `bt_str.c` it is easier to convert it to a unit test